subsys/bluetooth/mesh/adv.c

```diff
@@ -210,18 +234,24 @@ void bt_mesh_adv_send(struct net_buf *buf, const struct bt_mesh_send_cb *cb,
 	BT_MESH_ADV(buf)->cb_data = cb_data;
 	BT_MESH_ADV(buf)->busy = 1U;
 
+#if defined(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE)
+	if (BT_MESH_ADV(buf)->tag == BT_MESH_FRIEND_ADV) {
```

Comment:
`bt_mesh_friend_queue` and `bt_mesh_adv_buf_friend_ready` are always compiled, so IS_ENABLED should work here.